tools/Az.Tools.Predictor/Az.Tools.Predictor.SourceGenerator/SettingsSourceGenerator.cs

```diff
@@ -0,0 +1,199 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.CodeAnalysis;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+
+namespace Az.Tools.Predictor.SourceGenerator;
+
+sealed class SettingProperty
+{
+    public string Name { get; set; }
+
+    public string Type { get; set; }
+
+    public string Value { get; set; }
+}
+
+/// <summary>
+/// The source generator to create the part of the Settings class from the json file.
+///
+/// It gathers the key:value pair from the json file, and generates the public properties accordingly.
+/// The key from the json is the property name and the value from the json is the property value. It infers
+/// the property type based on the value.
+/// The generated properties are the default values of the settings.
+/// In addition to generating the properties, it also generate the method to get the settings from the user's profile,
+/// and override the property accordingly if it's set in the user's profile setting.
+/// </summary>
+[Generator]
+public class SettingsSourceGenerator : ISourceGenerator
```

Comment:
I just learned about this feature of dotnet. Thanks for doing this, really cool! :)